swarm-cluster-using-docker-extension/azuredeploy.json

```diff
@@ -0,0 +1,396 @@
+			"dnsSettings": {
+				"domainNameLabel": "[concat(parameters('dnsNameForPublicIP'), 'master')]"
+			}
+		}
+	},
+	{
+		"apiVersion": "2015-05-01-preview",
+		"type": "Microsoft.Network/publicIPAddresses",
+		"name": "[concat(variables('publicIPAddressName'), copyIndex())]",
+		"location": "[variables('location')]",
+		"copy": {
+			"name": "pubIpLoop",
+			"count": "[parameters('slaveCount')]"
+		},
+		"properties": {
+			"publicIPAllocationMethod": "[variables('publicIPAddressType')]",
+			"dnsSettings": {
+				"domainNameLabel": "[concat(parameters('dnsNameForPublicIP'), copyIndex())]"
+			}
+		}
+	},
+        {
+              "apiVersion": "2015-05-01-preview",
+              "type": "Microsoft.Network/virtualNetworks",
+              "name": "[variables('virtualNetworkName')]",
+              "location": "[variables('location')]",
+              "dependsOn": [
+			"[variables('nsgID')]"
+		],
+              "properties": {
+                "addressSpace": {
+                  "addressPrefixes": [
+                    "[variables('addressPrefix')]"
+                  ]
+                },
+                "subnets": [
+                  {
+                    "name": "[variables('subnetName')]",
+                    "properties" : {
+                      "addressPrefix": "[variables('subnetPrefix')]",
+		      "networkSecurityGroup" : {
+		              "id": "[variables('nsgID')]"
+			}
+                    }
+                  }
+                ]
+               }
+        },
+	{
+		"apiVersion": "2015-05-01-preview",
+		"type": "Microsoft.Network/networkSecurityGroups",
+		"name": "[variables('nsgName')]",
+		"location": "[variables('location')]",
+		"properties": {
+			"securityRules": [
+				{
+					"name": "swarmManagementPort",
+					"properties": {
+						"description": "Docker swarm endpoint",
+						"protocol": "Tcp",
+						"sourcePortRange": "*",
+						"destinationPortRange": "5555",
+						"sourceAddressPrefix": "*",
+						"destinationAddressPrefix": "*",
+						"access": "Allow",
+						"priority": 100,
+						"direction": "Inbound"
+					}
+				},
+				{
+					"name": "SSH",
+					"properties": {
+						"description": "SSH endpoint",
+						"protocol": "Tcp",
+						"sourcePortRange": "*",
+						"destinationPortRange": "22",
+						"sourceAddressPrefix": "*",
+						"destinationAddressPrefix": "*",
+						"access": "Allow",
+						"priority": 200,
+						"direction": "Inbound"
+					}
+				}
+			]
+		}
+	},
+	{
+		"apiVersion": "2015-05-01-preview",
+		"type": "Microsoft.Network/networkInterfaces",
+		"name": "[concat(variables('nicName'), 'master')]",
+		"location": "[variables('location')]",
+		"dependsOn": [
+			"[concat('Microsoft.Network/publicIPAddresses/', variables('publicIPAddressName'), 'master')]",
+			"[concat('Microsoft.Network/virtualNetworks/', variables('virtualNetworkName'))]"
+		],
+		"properties": {
+			"ipConfigurations": [
+				{
+					"name": "ipconfig1",
```

Comment:
instead of using suffixes like 1, 2 try use clearer names like ipConfigForMaster